src/host/notifyconnections.h

```diff
@@ -3,62 +3,132 @@
 #pragma once
 
 #include "ds/messaging.h"
+#include "everyio.h"
 #include "tcp.h"
 
+#include <curl/curl.h>
+
 namespace asynchost
 {
-  class NotifyConnections
+  class NotifyConnectionsImpl
   {
   private:
-    TCP notify_client;
-    bool is_setup = false;
+    CURLM* multi_handle = nullptr;
+    curl_slist* headers = nullptr;
 
-    class ClientBehaviour : public TCPBehaviour
-    {
-    private:
-      NotifyConnections& parent;
+    std::string notify_destination = {};
 
-    public:
-      ClientBehaviour(NotifyConnections& parent) : parent(parent) {}
+    // To ensure we cleanup all open handles, keep track of them locally
+    std::set<CURL*> easy_handles;
 
-      void on_resolve_failed()
+    void send_notification(const std::vector<uint8_t>& body)
+    {
+      if (multi_handle)
       {
-        LOG_DEBUG_FMT("notify client resolve failed");
-        reconnect();
+        CURL* curl = curl_easy_init();
+        easy_handles.insert(curl);
+
+        curl_easy_setopt(curl, CURLOPT_URL, notify_destination.c_str());
+        curl_easy_setopt(curl, CURLOPT_POST, 1L);
+        curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, body.size());
+        curl_easy_setopt(curl, CURLOPT_COPYPOSTFIELDS, body.data());
+
+        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+
+        curl_multi_add_handle(multi_handle, curl);
       }
```

Comment:
Minor: Should we log a failure if a notification is emitted from the enclave but cannot be curl-ed as the `multi_handle` wasn't set properly?

Reply:
I don't think we should log errors here - this is a legitimate case if the host was empty. Previously we used the `is_setup` bool. I'll add an error message for the case where `curl_multi_init` fails.